BLUESPAWN-client/src/util/filesystem/FileSystem.cpp

```diff
@@ -1,57 +1,223 @@
 #include "util/filesystem/FileSystem.h"
-
-bool CheckFileExists(LPCWSTR filename) {
+#include "util/log/Log.h"
+using namespace FileSystem; 
+bool FileSystem::CheckFileExists(LPCWSTR filename) {
 	//Function from https://stackoverflow.com/a/4404259/3302799
-	GetFileAttributesW(filename); 
-	if (INVALID_FILE_ATTRIBUTES == GetFileAttributes(filename) && 
+	GetFileAttributesW(filename);
+	if (INVALID_FILE_ATTRIBUTES == GetFileAttributes(filename) &&
 		GetLastError() == ERROR_FILE_NOT_FOUND)
 	{
+		LOG_VERBOSE(3, "File " << filename << " does not exist.");
 		return false;
 	}
+	LOG_VERBOSE(3, "File " << filename << " exists");
 	return true;
 }
 
-string GetFileContents(LPCWSTR filename) {
-	if (CheckFileExists(filename)) {
-		ifstream ifs(filename);
-		string content((std::istreambuf_iterator<char>(ifs)),
-			(std::istreambuf_iterator<char>()));
-		return content;
+void FileSystem::File::TranslateLongToFilePointer(long val, LONG& lowerVal, LONG& upperVal, PLONG& upper) {
+	//Calculate the offset into format needed for SetFilePointer
+	long lowerMask = 0xFFFFFFFF;
+	long upperMask = (long)(0x7FFFFFFF) << 32;
+	lowerVal = val & lowerMask;
+	upperVal = val & upperMask;
+	upper = &upperVal;
+}
+
+std::string WidestringToString(const std::wstring& wstr) {
+	int size = 0;
+	WideCharToMultiByte(CP_ACP, 0, wstr.c_str(), static_cast<int>(wstr.length()), nullptr, 0, nullptr, &size);
+	CHAR* str = new CHAR[size];
+	WideCharToMultiByte(CP_ACP, 0, wstr.c_str(), static_cast<int>(wstr.length()), str, size, nullptr, nullptr);
+	return str;
+}
+
+FileSystem::File::File(IN const LPCWSTR path) {
+	FilePath = path;
+	LOG_VERBOSE(2, "Attempting to open file: " << path << ".");
+	hFile = CreateFileW(path, 
+		GENERIC_READ | GENERIC_WRITE, 
+		FILE_SHARE_READ,
+		NULL, 
+		OPEN_EXISTING,
+		FILE_FLAG_SEQUENTIAL_SCAN | FILE_ATTRIBUTE_NORMAL,
+		NULL);
+	if (INVALID_HANDLE_VALUE == hFile)
+	{
+		DWORD dwStatus = GetLastError();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'dwStatus' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4959972)